src/authority.rs

```diff
@@ -82,7 +89,9 @@ pub fn our_authority(message       : &RoutingMessage,
         MessageType::ConnectResponse(_)     => None,
         MessageType::FindGroup              => None,
         MessageType::FindGroupResponse(_)   => None,
-        MessageType::GetData(_)             => Some(message.non_relayed_destination()),
+        MessageType::GetData(_)             => {
+            Some(message.destination().get_location().clone())
```

Comment:
I think so. IIRC in pre-rust-2 sprint code we had `GetData.data.name()` same as the destination the same, and it was carried here with it.